components/hab/src/license.rs

```diff
@@ -128,17 +128,29 @@ pub fn accept_license(ui: &mut UI) -> Result<()> {
     Ok(())
 }
 
-fn license_path() -> PathBuf {
-    let hab_dir = if am_i_root() {
-        PathBuf::from(&*FS_ROOT_PATH).join("hab")
-    } else if let Some(home) = dirs::home_dir() {
+fn superuser_license_root() -> PathBuf { PathBuf::from(&*FS_ROOT_PATH).join("hab") }
+
+fn user_license_root() -> PathBuf {
+    if let Some(home) = dirs::home_dir() {
         home.join(".hab")
     } else {
         panic!("No home directory available. Unable to find a suitable place to write a license \
```

Comment:
Just a question on whether panic is what we want here.. I realize its code thats not specifically for this change but it seems odd to have a panic in that path

Reply:
Yeah the panic is intentional.  If we hit that code path, something is _super_ wrong because it means that `hab` is being run by a user that doesn't have a home directory.  Users without home directories tend to be ones that are reserved for running services and as such, wouldn't be appropriate for accepting a license, since there's no sensible place to persist the license acceptance file.